### PR TITLE
DDF-2498 Added ability to specify source and target for Migrate Command

### PR DESCRIPTION
**Pull request by emmberk**
#### What does this PR do?
This PR adds `--from` and `--to` options to the Migrate Command to specify the providers, and it adds a `--list` option to list the available providers.
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@bdeining @glenhein @peterhuffer 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@beyelerb
@kcwire
#### How should this be tested?
Add an external provider, and test that the `catalog:migrate` and the options work as expected when migrating from this new external provider to the `SolrHttpCatalogProvider`.
#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2498
#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java [hunk 1]
```diff
@@ -43,44 +47,66 @@
...
+        if (listProviders) {
+            if (providers.size() == 0) {
+                console.println("There are no available providers.");
+                return null;
+            }
+            console.println("Available providers:");
+            providers.stream()
+                    .map(p -> p.getClass()
+                            .getSimpleName())
+                    .forEach(id -> console.println("\t" + id));
```
Comment (peterhuffer) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java]: Should we return on this and end the operation when the `--list` option is used? I can't imagine something like `catalog:migrate --list -to x -from y` being done.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java [hunk 2]
```diff
@@ -43,44 +47,66 @@
...
-        + "from a Configured External Provider into the Catalog.  The migrate command currently "
-        + "picks the first Catalog provider as the FROM and the second Catalog provider as the TO. "
-        + "Additional support will be added to specify which provider will be migrated TO and FROM "
-        + "in the future.")
+        + "from a Configured External Provider to another Provider.")
 public class MigrateCommand extends DuplicateCommands {
     private static final Logger LOGGER = LoggerFactory.getLogger(MigrateCommand.class);
 
+    @Option(name = "--list", required = false, aliases = {
+            "-list"}, multiValued = false, description = "Print a list of external providers before migrating.")
```
Comment (peterhuffer) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java]: Is this necessarily only printing external providers? Won't this `--list` option also print the default local provider? Similar comment for the `--from` and `--to` options.

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java (hunk 1 above)
Comment (emmberk) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java]: Updated to complete command after listing providers.

**Comment by glenhein** [comment]
Emily suggested two different approaches to hero the code. I used the approach where I commented out the code that enforces the requirement that there are at least two providers. That way I could migrate to/from the same instance.

`ddf@local>catalog:migrate -from SolrHttpCatalogProvider -to SolrHttpCatalogProvider
FROM provider ID: SolrHttpCatalogProvider
TO provider ID: SolrHttpCatalogProvider
Starting migration.
 100% [===================================================>]     6 records/sec	
 2 record(s) migrated; 0 record(s) failed; completed in 0.305 seconds.
`

File: catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java [hunk 3]
```diff
@@ -43,44 +47,68 @@
...
+            providers.stream()
+                    .map(p -> p.getClass()
+                            .getSimpleName())
+                    .forEach(id -> console.println("\t" + id));
+
+            return null;
+        }
+
         if (batchSize > MAX_BATCH_SIZE || batchSize < 1) {
             console.println("Batch Size must be between 1 and 1000.");
```
Comment (beyelerb) [inline catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/MigrateCommand.java]: use MAX_BATCH_SIZE here?

**Pull request merged**